### PR TITLE
Initial gradle build/tests and boilerplate for plugin

### PR DESCRIPTION
**Pull request by dbbaughe**
*Issue #, if available:*

*Description of changes:*
Sets up the initial plugin for Index State Management

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+sourceCompatibility = targetCompatibility = "1.8"
+compileKotlin.kotlinOptions.jvmTarget = compileTestKotlin.kotlinOptions.jvmTarget = "1.8"
+
+javadoc.enabled = false // turn off javadoc as it barfs on Kotlin code
+licenseHeaders.enabled = true
+dependencyLicenses.enabled = false
+thirdPartyAudit.enabled = false
+
+unitTest {
+    systemProperty 'tests.security.manager', 'false'
```
Comment (zengyan-amazon) [inline build.gradle]: why this is needed for unit test?

File: settings.gradle [hunk 2]
```diff
@@ -0,0 +1 @@
+rootProject.name = 'opendistro-ism'
```
Comment (zengyan-amazon) [inline settings.gradle]: shall we use a more self-explain name?

Comment (zengyan-amazon) [reply settings.gradle]: also add license header in this file

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,77 @@
+buildscript {
```
Comment (zengyan-amazon) [inline build.gradle]: shall we add license header to this file?

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    ext {
+        es_version = System.getProperty("es.version", "6.7.1")
+        kotlin_version = System.getProperty("kotlin.version", "1.3.31")
```
Comment (jinsoor-amzn) [inline build.gradle]: Lets add Kotlin Linter as early in the development phase as we can to minimize code style differences.

Comment (dbbaughe) [reply build.gradle]: Yep, will do in a follow up PR; there is still quite a bit to add to the gradle configuration which won't be covered by this PR.

File: build.gradle (hunk 1 above)
Comment (dbbaughe) [reply build.gradle]: Not needed, removed

File: settings.gradle (hunk 2 above)
Comment (dbbaughe) [reply settings.gradle]: Added more descriptive name and copyright

File: build.gradle (hunk 3 above)
Comment (dbbaughe) [reply build.gradle]: Added

**Pull request merged**